### ISSUE TITLE
Subtitle plugin API requests

### ISSUE THREAD

**Issue by Ami-OS** [Pong420/videojs-plus#18]
Hi, i need to get some values from Subtitle plugin to implement some functions, but the existing api cannot get the value I need (Or i don't know),
help ~~

### I need to get the following value or status :

- [status] Whether subtitles are on or off
- [value] Get current subtitles index id (Is what `pick(index)` Really needed)
- [value] Get default subtitles index id (`default: true`)

`player.subtitles().track` corresponding index id, for use `player.subtitles().pick(index);`

Thx

**Comment by Pong420** [comment]
`player.subtitles().values()` will return a list of `TextTrack`.

> [status] Whether subtitles are on or off

you may loop through the return array and get textrack `mode`, `showing` is on and `hidden` is off
```js
player.subtitles().values().map(t => t.mode)
 ```
---

> [value] Get current subtitles index id (Is what pick(index) Really needed);
```js
player.subtitles().values().indexOf(player.subtitles().track)
```
or subscribe `subtitlechange`
```js
player.on('subtitlechange', (event,{index, label}) => console.log(index, label))
```
---

> [value] Get default subtitles index id (default: true)

I don't understand. The subtitles options are handled by yourself. You should know the index.

Maybe you could 
```js
player.options().subtitles.findIndex((option) => option.default === true)
```
or
```js
player.on('subtitles', (event, subtitles) => {
   console.log('subtitles setup', subtitles);
});
```

**Comment by Ami-OS** [comment]
I spent some time researching, These are exactly what i want!
I also found a special thing:

`player.subtitles().values().indexOf(player.subtitles().track)`

This will get the current subtitle index id except "-1", This has a pros and a cons

A pros:

This can get the user's last selected subtitle index id.

A cons:

This cannot get the most real-time status.

So I asked the following questions:

> [status] Whether subtitles are on or off

Then I combined the answer to the first question.
Get:

`player.subtitles().values().map(t => t.mode).indexOf('showing')`

This will look for a specific state 'showing' in the array, if match will output index id and if it can’t match, it will output `-1`,
this is exactly the closed state.

---

I have implemented some hotkey functions using the above api.
thank you very much.
